src/backend/tests/routers/test_agent.py

```diff
@@ -971,6 +1077,62 @@ def test_update_agent_invalid_tool(
     assert response.json() == {"detail": "Tool not a real tool not found."}
 
 
+def test_update_private_agent(
+    session_client: TestClient, session: Session, user
+) -> None:
+    agent = get_factory("Agent", session).create(
+        name="test agent",
+        version=1,
+        description="test description",
+        preamble="test preamble",
+        temperature=0.5,
+        model="command-r-plus",
+        deployment=ModelDeploymentName.CoherePlatform,
+        is_private=True,
+        user_id=user.id,
+    )
+
+    request_json = {
+        "name": "updated name",
+    }
+
+    response = session_client.put(
+        f"/v1/agents/{agent.id}", json=request_json, headers={"User-Id": user.id}
+    )
+    assert response.status_code == 200
+    updated_agent = response.json()
+    assert updated_agent["name"] == "updated name"
+    assert updated_agent["is_private"] == True
+
+
+def test_update_public_agent(
+    session_client: TestClient, session: Session, user
+) -> None:
+    agent = get_factory("Agent", session).create(
+        name="test agent",
+        version=1,
+        description="test description",
+        preamble="test preamble",
+        temperature=0.5,
+        model="command-r-plus",
+        deployment=ModelDeploymentName.CoherePlatform,
+        is_private=False,
+        user_id=user.id,
+    )
+
+    request_json = {
+        "name": "updated name",
+    }
+
+    response = session_client.put(
+        f"/v1/agents/{agent.id}", json=request_json, headers={"User-Id": user.id}
+    )
+    assert response.status_code == 200
+    updated_agent = response.json()
+    assert updated_agent["name"] == "updated name"
+    assert updated_agent["is_private"] == False
+
+
```

Comment:
Can we update a private Agent to public and vice versa? Might be good to add some test scenarios for those (eg, a private -> public agent would now be accessible through crud by any user)

Reply:
If no bandwidth feel free to make a ticket a prioritize as needed, not necessary to get in but would be good coverage

Reply:
Done!